File: src/types/recipe.ts
export interface Recipe {
  id: number;
  name: string;
  region: string;
  difficulty: 'Easy' | 'Medium' | 'Hard';
  cookTime: string;
  image: string;

  // Optional detailed properties
  description?: string;
  prepTime?: string;
  servings?: number;
  calories?: number;
  ingredients?: string[];
  instructions?: string[];
  nutrition?: {
    calories?: number;
    protein?: string;
    carbs?: string;
    fat?: string;
    fiber?: string;
  };
  tags?: string[];
  
  // Optional UI properties
  cuisine?: string;
  rating?: number;
  source?: 'database' | 'imported' | 'custom';
  createdAt?: string;
}

export interface RecipeDatabase {
  [cuisine: string]: Recipe[];
}

// The detail part of a recipe
export type RecipeDetail = Omit<Recipe, 'id' | 'name' | 'region' | 'difficulty' | 'cookTime' | 'image' | 'cuisine' | 'rating'>;

export interface RecipeDetailsDatabase {
  [id: string]: RecipeDetail;
}

File: minimal_chef/lib/models/recipe.dart
import 'package:minimal_chef/models/ingredient.dart';

class Recipe {
  final String id;
  final String name;
  final String description;
  final String imageUrl;
  final List<String> instructions;
  final List<Ingredient> ingredients;

  Recipe({
    required this.id,
    required this.name,
    required this.description,
    required this.imageUrl,
    required this.instructions,
    required this.ingredients,
  });
}